ssl/record/methods/dtls_meth.c

```diff
@@ -170,8 +170,7 @@ static int dtls_process_record(OSSL_RECORD_LAYER *rl, DTLS_BITMAP *bitmap)
         mac = rr->data + rr->length;
         i = rl->funcs->mac(rl, rr, md, 0 /* not send */);
         if (i == 0 || CRYPTO_memcmp(md, mac, (size_t)mac_size) != 0) {
-            RLAYERfatal(rl, SSL_AD_BAD_RECORD_MAC,
-                        SSL_R_DECRYPTION_FAILED_OR_BAD_RECORD_MAC);
+            /* Silently drop the packet in case of invalid MAC. */
```

Comment:
Should also the RLAYERfatal() call at line 166 be dropped? It is just another way how the packet can be corrupted.

Reply:
The RFC mentions length issues in the same sentence so shouldn't we handle length issues the same way invalid macs? Here are some examples from `dtls_process_record()`:

https://github.com/openssl/openssl/blob/bbe8ab29e1cf394a978df29f04695f56f1b166cf/ssl/record/methods/dtls_meth.c#L140-L143

https://github.com/openssl/openssl/blob/bbe8ab29e1cf394a978df29f04695f56f1b166cf/ssl/record/methods/dtls_meth.c#L236-L241

I'm not sure if max_frag_len MUST be handled with a fatal alert?
https://github.com/openssl/openssl/blob/bbe8ab29e1cf394a978df29f04695f56f1b166cf/ssl/record/methods/dtls_meth.c#L252-L255

Reply:
@fwh-dc I wonder what about this alert:
https://github.com/openssl/openssl/blob/bbe8ab29e1cf394a978df29f04695f56f1b166cf/ssl/record/methods/dtls_meth.c#L242-L244
The RFC didn't explicitly mention decompression failures, but it only tried to give examples for invalid records and did not list all possible reasons - "DTLS is resilient in the face of invalid records (e.g., invalid formatting, length, MAC, etc.)".
I think decompression failure could be considered "invalid record", and it is the only fatal error left in this file that is caused by the record data and not an internal error.

Reply:
This section of TLS 1.2 rfc describes a fatal warning:
https://datatracker.ietf.org/doc/html/rfc5246#section-6.2.2
`tls_do_uncompress()` returns 0 in case of any failure. So we should keep this fatal alert if I understand the code correctly.

Reply:
Sounds right, thank you